flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/consumer/TestInputGate.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.io.network.partition.consumer;
+
+import org.apache.flink.runtime.clusterframework.types.ResourceID;
+import org.apache.flink.runtime.event.TaskEvent;
+import org.apache.flink.runtime.io.network.buffer.BufferPool;
+import org.apache.flink.runtime.io.network.buffer.BufferProvider;
+import org.apache.flink.runtime.io.network.buffer.BufferReceivedListener;
+import org.apache.flink.runtime.io.network.partition.ResultPartitionID;
+import org.apache.flink.runtime.io.network.partition.ResultPartitionType;
+import org.apache.flink.runtime.jobgraph.IntermediateResultPartitionID;
+import org.apache.flink.runtime.shuffle.NettyShuffleDescriptor;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Timer;
+import java.util.concurrent.CompletableFuture;
+
+class TestInputGate implements SingleInputGate {
```

Comment:
Why do you need this class? Couldn't you use:
```
new SingleInputGateBuilder().setNumberOfChannels(numberOfChannels).build();
```
?

Besides `TestInputGate` being a bit dangerous - some methods are implemented incorrectly without throwing an exception, it will add quite a bit of extra maintenance whenever we will going to change `SingleInputGate` interface.

Reply:
Reverted to minimize changes in this PR.